### PR TITLE
Update README.md and environment.yml

### PR DESCRIPTION
**Pull request by mwcvitkovic**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,14 @@
 # ML Reproducibility Challenge 2020: Your Classifier is Secretly an Energy Based Model and You Should Treat it Like One
 
-This repo contains a re-implemimentation of the above paper https://arxiv.org/abs/1912.03263. Local code was run on Mac OS X Mojave, version 10.14.6
-## Installation/dependencies
+This repo contains a re-implemimentation of the 2020 ICLR paper [Your Classifier is Secretly an Energy Based Model and You Should Treat it Like One](https://arxiv.org/abs/1912.03263).
```
Comment (mwcvitkovic) [inline README.md:3]: Links in markdown are `[words](url)`

File: README.md [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
+Local code was run on Mac OS X Mojave, version 10.14.6
 
-1. Install [conda](https://docs.anaconda.com/anaconda/install/)
-2. `python==3.7.7`
-3. `torch==1.6.0`
-4. `torchvision==0.7.0`
-5. `tqdm==4.47.0`
```
Comment (mwcvitkovic) [inline README.md:10]: You want to let the environment.yml file specify all this

Comment (mwcvitkovic) [inline README.md:10]: Also weirdly in markdown, to make a list it's better to start every line with `1.`

File: README.md [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
+## Installation
 
+1. Install [conda](https://docs.anaconda.com/anaconda/install/)
+1. Create conda environment:
+    `conda env create -f environment.yml`
+1. Activate environment
+    `conda activate ml_reprod_hybrid_energy_models`
+1. Confirm you can run the training script
+    `python train_supervised.py`
```
Comment (mwcvitkovic) [inline README.md:14]: The ideal installation instructions for any repo are basically a series of commands I can copy and paste right into the command line.

This ends up being for our own sake 99% of the time ;)

File: environment.yml [hunk 4]
```diff
@@ -1,319 +1,9 @@
-name: pytorch_updated
+name: ml_reprod_hybrid_energy_models
```
Comment (mwcvitkovic) [inline environment.yml:1]: Usually better to give the env a semantic name

File: environment.yml [hunk 5]
```diff
@@ -1,319 +1,9 @@
...
 channels:
   - pytorch
   - defaults
 dependencies:
-  - alabaster=0.7.12=py37_0
```
Comment (mwcvitkovic) [inline environment.yml:6]: Even though conda exports every exact library you used, it's better for portability to only specify the libraries you need.

In your case these are the ones you had in the README.  But they should live in `environment.yml`

File: environment.yml [hunk 6]
```diff
@@ -1,319 +1,9 @@
...
-    - murmurhash==1.0.2
-    - plac==1.1.3
-    - preshed==3.0.2
-    - sentencepiece==0.1.91
-    - spacy==2.3.2
-    - srsly==1.0.2
-    - thinc==7.4.1
-    - torchtext==0.7.0
-    - wasabi==0.8.0
-prefix: /Users/DivyanshuMurli1/anaconda3/envs/pytorch_updated
```
Comment (mwcvitkovic) [inline environment.yml:319]: Best to keep this line out - it contains specific info about your computer filesystem, which is kindof private

File: environment.yml [hunk 7]
```diff
@@ -1,319 +1,9 @@
...
-  - pysocks=1.7.1=py37_0
-  - pytables=3.6.1=py37h5bccee9_0
-  - pytest=5.4.3=py37_0
-  - python=3.7.7=hf48f09d_4
-  - python-dateutil=2.8.1=py_0
-  - python-jsonrpc-server=0.3.4=py_1
-  - python-language-server=0.34.1=py37_0
-  - python-libarchive-c=2.9=py_0
-  - python.app=2=py37_10
-  - pytorch=1.6.0=py3.7_0
```
Comment (mwcvitkovic) [inline environment.yml:216]: Also best to avoid the specific python version (the `py3.7_0` part) and let conda figure that out

**Review by divymurli — APPROVED** [review]

**Pull request merged**